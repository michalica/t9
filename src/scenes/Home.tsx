import React from 'react';
import UserInput from '../components/UserInput';
import styled from 'styled-components';
import { Container } from 'semantic-ui-react'

const Logo = styled.h1`
    font-size: 22px
    font-weight: bold
    margin-top: 80px!important
`

function Home({}) {
    return (
        <div>
            <Container textAlign='center'>
                <Logo>T9</Logo>
                <UserInput />
            </Container>
        </div>
    );
};

export default Home;
