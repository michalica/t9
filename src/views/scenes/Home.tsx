import React from 'react';
import UserInput from '../components/UserInput';
import Result from '../components/Result';

import styled from 'styled-components';
import {Container} from 'semantic-ui-react'
import Keyboard from '../components/Keyboard';

const Logo = styled.h1`
    font-size: 22px
    font-weight: bold
    margin-top: 80px!important
`

interface PropsInterface {
    fetchApi: Function,
    letters: Array<string>,
    isLoading: boolean,
    validationError: boolean,
    formValue: string,
}

class Home extends React.Component<PropsInterface, any> {
    constructor(props: PropsInterface) {
        super(props);
    }

    handleChange = (values: String) => {
        this.props.fetchApi(values);
        console.log(values);
    }

    render() {
        const {
            letters,
            isLoading,
            validationError,
            formValue,

        } = this.props;
        return (
            <div>
                <Container textAlign='center'>
                    <Logo>T9</Logo>
                    <UserInput
                        isLoading={isLoading}
                        handleChange={this.handleChange}
                        validationError={validationError}
                        formValue={formValue}
                        type="number"
                        placeholder="Search for ..."
                    />
                    <br/>
                    <br/>
                    <Keyboard
                        handleChange={this.handleChange}
                        value={formValue}
                    />
                    <Result
                        letters={letters}
                    />
                </Container>
            </div>
        );
    }
};

export default Home;
