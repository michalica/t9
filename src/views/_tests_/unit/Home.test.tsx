import React from 'react';
import renderer from "react-test-renderer";
import Home from '../../scenes/Home';

const fethcApiHandler = jest.fn();
const doRequestHandler = jest.fn();

const props = {
    fetchApi: fethcApiHandler,
    letters: ['a','b'],
    isLoading: true,
    validationError: false,
    formValue: '123',
    doRequest: doRequestHandler,
}

it('render withou crasing', () => {
    const tree = renderer
        .create(<Home {...props} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
