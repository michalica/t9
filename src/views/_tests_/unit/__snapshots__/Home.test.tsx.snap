// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render withou crasing 1`] = `
<div>
  <div
    className="ui center aligned container"
  >
    <h1
      className="sc-bxivhb cRJeLh"
    >
      T9
    </h1>
    <div
      className="ui loading icon input"
    >
      <input
        onChange={[Function]}
        placeholder="Search for ..."
        type="number"
        value="123"
      />
      <i
        aria-hidden="true"
        className="spinner icon"
        onClick={[Function]}
      />
    </div>
    <br />
    <br />
    <div>
      <div>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          1
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          2
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          3
        </button>
        <br />
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          4
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          5
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          6
        </button>
        <br />
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          7
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          8
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          9
        </button>
        <br />
        <button
          className="ui icon button sc-htpNat iyezmu"
          disabled={true}
        >
          #
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          0
        </button>
        <button
          className="ui icon button sc-htpNat iyezmu"
          onClick={[Function]}
        >
          &lt;
        </button>
      </div>
    </div>
    <div>
      <ul
        className="sc-bdVaJa loehwO"
      >
        <li
          className="sc-bwzfXH erLWkn"
        >
           
          a
           
        </li>
        <li
          className="sc-bwzfXH erLWkn"
        >
           
          b
           
        </li>
      </ul>
    </div>
  </div>
</div>
`;
